connect/runtime/src/main/java/org/apache/kafka/connect/runtime/Worker.java

```diff
@@ -282,11 +337,14 @@ public boolean startTask(
             return false;
         }
 
+        WorkerTask existing = tasks.put(id, workerTask);
```

Comment:
After using the `ConcurrentMap` interface instead, this is the reasonable thing to do (`putIfAbsent`), although we throw an exception anyways.